pkg/operator/ceph/object/config.go

```diff
@@ -111,3 +111,24 @@ func (c *clusterConfig) setDefaultFlagsMonConfigStore(rgwName string) error {
 
 	return nil
 }
+
+func (c *clusterConfig) deleteFlagsMonConfigStore(rgwName string) error {
+	monStore := cephconfig.GetMonStore(c.context, c.store.Namespace)
+	who := generateCephXUser(rgwName)
+	configOptions, err := monStore.GetDaemon(who)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get daemon config for %q", who)
+	}
+
+	for _, option := range configOptions {
+		// Only remove specialized options (don't remove global one)
+		if option.Who == who {
```

Comment:
Why don't we do this in `GetDaemon` directly since it's supposed to return the config of a particular daemon? Also, we would need to implement this for other daemons too so replicating this pattern is not needed.

Reply:
My first idea was to provide a generic function returning all flags of a daemon as the ceph command return it, just in case some rook could need it for other use case.
But it is probably useless, agree that we can move this to the `GetDaemon` and should also create a DeleteDaemon that will delete all options for a daemon